stdlib/public/Darwin/Accelerate/vDSP_Conversion.swift

```diff
@@ -0,0 +1,796 @@
+    @inline(__always)
+    @available(iOS 9999, OSX 9999, tvOS 9999, watchOS 9999, *)
+    public static func convert<U, V>(_ source: U,
+                                     to destination: inout V,
+                                     rounding: RoundingMode)
+        where
+        U: _ContiguousCollection,
+        V: _MutableContiguousCollection,
+        U.Element == Float,
+        V.Element == Int8 {
+            let n = vDSP_Length(min(source.count,
+                                    destination.count))
+            
+            destination.withUnsafeMutableBufferPointer { dest in
+                source.withUnsafeBufferPointer { src in
+                    switch rounding {
+                    case .towardsZero:
+                        vDSP_vfix8(src.baseAddress!, 1,
+                                   dest.baseAddress!, 1,
+                                   n)
+                    case .towardsNearestInteger:
+                        vDSP_vfixr8(src.baseAddress!, 1,
+                                    dest.baseAddress!, 1,
+                                    n)
+                    }
+                }
+            }
+    }
+    
+    /// Converts an array of double-precision floating-point values to signed 8-bit integer values using specified rounding.
+    ///
+    /// - Parameter source: Source vector.
+    /// - Parameter destination: Destination vector.
+    @inline(__always)
+    @available(iOS 9999, OSX 9999, tvOS 9999, watchOS 9999, *)
+    public static func convert<U, V>(_ source: U,
+                                     to destination: inout V,
+                                     rounding: RoundingMode)
+        where
+        U: _ContiguousCollection,
+        V: _MutableContiguousCollection,
+        U.Element == Double,
+        V.Element == Int8 {
+            let n = vDSP_Length(min(source.count,
+                                    destination.count))
+            
+            destination.withUnsafeMutableBufferPointer { dest in
+                source.withUnsafeBufferPointer { src in
+                    switch rounding {
+                    case .towardsZero:
+                        vDSP_vfix8D(src.baseAddress!, 1,
+                                    dest.baseAddress!, 1,
+                                    n)
+                    case .towardsNearestInteger:
+                        vDSP_vfixr8D(src.baseAddress!, 1,
+                                     dest.baseAddress!, 1,
+                                     n)
+                    }
+                }
+            }
+    }
+    
+    /// Converts an array of single-precision floating-point values to unsigned 8-bit integer values using specified rounding.
+    ///
+    /// - Parameter source: Source vector.
+    /// - Parameter destination: Destination vector.
+    @inline(__always)
+    @available(iOS 9999, OSX 9999, tvOS 9999, watchOS 9999, *)
+    public static func convert<U, V>(_ source: U,
```

Comment:
There's a lot of repetition here, I wonder whether it would be possible to avoid that with a clever generic function...

```swift

typealias ConversionFunc<U, V> = (UnsafePointer<U>, Int, UnsafeMutablePointer<V>, Int, UInt) -> Void
private static func convertImpl<U, V>(
  _ source: U,
  to destination: inout V,
  rounding: RoundingMode,
  _ towardZeroFunc: ConversionFunc<U, V>,
  _ towardNearestFunc: ConversionFunc<U, V> 
)
where
  U: _ContiguousCollection,
  V: _MutableContiguousCollection {
  let n = vDSP_Length(min(source.count, destination.count))

  destination.withUnsafeMutableBufferPointer { dest in
    source.withUnsafeBufferPointer { src in
      switch rounding {
      case .towardsZero:
        towardZeroFunc(src.baseAddress!, 1, dest.baseAddress!, 1,  n)
      case .towardsNearestInteger:
        towardNearestFunc(src.baseAddress!, 1, dest.baseAddress!, 1, n)
      }
    }
  }
}
```

and then some (all?) of the `convert` overloads will look something like:

```swift
public static func convert<U, V>(
  _ source: U,
  to destination: inout V,
  rounding: RoundingMode
) where
  U: _ContiguousCollection,
  V: _MutableContiguousCollection,
  U.Element == Float,
  V.Element == UInt8 {
  convertImpl(source, to: &destination, rounding: rounding, vDSP_vfixu8, vDSP_vfixru8)
}
```